xbmc/filesystem/SMBFile.cpp

```diff
@@ -151,6 +158,10 @@ void CSMB::Init()
 
     // setup our context
     m_context = smbc_new_context();
+
+    // restore HOME
+    setenv("HOME", truehome, 1);
```

Comment:
Yes, I've tried that as well but same result.

I added this patch for diagnostics: http://sprunge.us/HbHG
In LibreELEC I get:
```
21:14:10.706 T:1784505248    INFO: NJM3: $HOME is now [/storage/.kodi/]
21:14:10.708 T:1784505248    INFO: NJM4: $HOME needs to be [/storage]
21:14:10.709 T:1784505248    INFO: NJM5: $HOME has been set to [/storage], retcode = 0, errno = 95, error = Operation not supported
```
while in Ubuntu 16.04 it does appear to work:
```
21:57:31.969 T:140159918352128    INFO: NJM3: $HOME is now [/home/neil/.kodi/]
21:57:31.969 T:140159918352128    INFO: NJM4: $HOME needs to be [/home/neil]
21:57:31.969 T:140159918352128    INFO: NJM5: $HOME has been set to [/home/neil], retcode = 0, errno = 2, error = No such file or directory
```
so for now at least it's just an LE issue. Hmmm.